configcatclient/readwritelock.py

```diff
@@ -30,3 +30,75 @@ def acquire_write(self):
 
     def release_write(self):
         self.__exclude.release()
+
+
+class RWLock(object):
```

Comment:
I actually think this should replace the other lock. This concurrency issue isn't specific to the lazy loading. (is it?)

Reply:
Yeah, I guess the concern I had was that, we hadn't used those and I hadn't thoroughly investigated if they have the same issue or not.